### DISCUSSION TITLE
Is there a Google Animals like model selection bottom slide available?

### DISCUSSION THREAD

**Discussion by calin-ciobanu** [google/model-viewer#1892] [Q&A]
I see this new functionality when viewing animals from Google Search:
![tempFileForShare_20200609-0942412](https://user-images.githubusercontent.com/5766012/84097422-217d3b80-aa37-11ea-9855-8407e053460b.jpg)

Is this / will be available in the scene-viewer / model viewer?
I see no reference yet looking at:
https://developers.google.com/ar/develop/java/scene-viewer#3d-or-ar

**Comment by methot-ilya-3ds** [comment]
I've made something similar to the image that you are showing.

In my example I had the Google Model viewer on a separate page. I then included that page as an iframe and by using CSS design you can overlay buttons on top of the iframe.

The buttons can either change the object source of the widget, or, in my case I just created seperate webpages for each object and changed the individual iframe when someone clicked the buttons.

**Comment by elalish** [comment]
I have no idea if/when SceneViewer plans to make their carousel a public API - @tpsiaki? However, it is easy to construct one for model-viewer, as you can simply put extra HTML elements as children of the model-viewer element and use them to change the `src`. Using webXR for AR instead of SceneViewer (soon our default) allows any HTML customization to flow naturally into AR mode as well. I built of a demo of this which will be going live on our docs pages as soon as we ship v1.0, which should be imminent.

**Comment by hybridherbst** [comment]
@elalish just to get that right, WebXR will not work for the foreseeable future on iOS until Apple finally decides to jump on the train?

**Comment by elalish** [comment]
@soraryu Indeed, and please feel free to tell apple how important webXR is to you! I've also seen things like [this](https://labs.mozilla.org/projects/webxr-viewer/) that look interesting.

**Comment by elalish** [comment]
You can now see my carousel [demo](https://modelviewer.dev/examples/webxr.html) on our dev pages, which will also be visible in AR on an ARCore capable Android device with the latest Chrome.

**Comment by hybridherbst** [comment]
Nice demo! Thank you very much!
Is scaling support for WebXR somewhere on the roadmap to 1.0?

**Comment by elalish** [comment]
@hybridherbst It has it already! I put scale="fixed" since these are shopping models, but maybe I should remove it. Or at least note it.

**Comment by elalish** [comment]
And 1.0 was just [released](https://github.com/google/model-viewer/releases/tag/v1.0.0)!

**Comment by calin-ciobanu** [comment]
thanks everyone and :tada: for the v1.0

**Comment by lupohirp** [comment]
@elalish i have a question/ though.  I'm developing a react  js app, and your library works great.  Howerer, as you can said previuosly, WebXR are not supported by Apple ( damn......). BUT, maybe we can use [this](https://github.com/immersive-web/webxr-polyfill)? Maybe we can integrate in our project and give the support for the missing APIs from the polyfills instead of the browser, like it happens with angular and the missing information on IE. In theory,  Safari will have the webxr APIs and your plugin will work with WebXR. I must try later in the evening....what do you think?

**Comment by hybridherbst** [comment]
@lupohirp Polyfills still require a supporting underlying API. The problem is that Apple supports neither WebVR nor WebAR nor anything else what would give access to AR features to the browser at this point. They are offloading all the XR work to their QuickLook viewer (a native app).
(elalish correct me if I'm wrong please)

**Comment by lupohirp** [comment]
What ??? The don't have even the underlyng api's? Why People buy this s*****t i will never understand. I thought that they had implemented but never exposed.....rhanks anyway

**Comment by elalish** [comment]
Possible hope for the future? https://bugs.webkit.org/show_bug.cgi?id=208988
